ansible/roles/invoker/tasks/deploy.yml

```diff
@@ -3,16 +3,40 @@
 
 - include: docker_login.yml
 
-- name: "pull action images with tag {{docker_image_tag}}"
+- name: "pull invoker images with tag {{docker_image_tag}}"
   shell: "docker pull {{docker_registry}}{{item}}:{{docker_image_tag}}"
   with_items:
     - '{{ docker_image_prefix }}/invoker'
-    - '{{ docker_image_prefix }}/dockerskeleton'
-    - '{{ docker_image_prefix }}/nodejs6action'
-    - '{{ docker_image_prefix }}/python2action'
-    - '{{ docker_image_prefix }}/python3action'
-    - '{{ docker_image_prefix }}/swift3action'
-    - '{{ docker_image_prefix }}/java8action'
+  when: docker_registry != ""
+
+- name: "pull nodejs action images with tag {{docker_image_tag}}"
+  shell: "docker pull {{docker_registry}}{{docker_image_prefix}}/{{item.image.name}}:{{docker_image_tag}}"
+  with_items:
+    - "{{ runtimesManifest.runtimes.nodejs }}"
+  when: (docker_registry != "") and ("deprecated" not in item)
+
+- name: "pull python action images with tag {{docker_image_tag}}"
+  shell: "docker pull {{docker_registry}}{{docker_image_prefix}}/{{item.image.name}}:{{docker_image_tag}}"
+  with_items:
+    - "{{ runtimesManifest.runtimes.python }}"
+  when: (docker_registry != "") and ("deprecated" not in item)
+
+- name: "pull swift action images with tag {{docker_image_tag}}"
+  shell: "docker pull {{docker_registry}}{{docker_image_prefix}}/{{item.image.name}}:{{docker_image_tag}}"
+  with_items:
+    - "{{ runtimesManifest.runtimes.swift }}"
+  when: (docker_registry != "") and ("deprecated" not in item)
+
+- name: "pull java action images with tag {{docker_image_tag}}"
+  shell: "docker pull {{docker_registry}}{{docker_image_prefix}}/{{item.image.name}}:{{docker_image_tag}}"
+  with_items:
+    - "{{ runtimesManifest.runtimes.java }}"
+  when: (docker_registry != "") and ("deprecated" not in item)
```

Comment:
As it stands right now, you could do `selectattr('deprecated', 'undefined')` because we haven't defined `deprecated: False` anywhere.

I'd prefer we just define the key everywhere and set it to False for completeness. I think that's less confusing for everyone WDYT?